src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs

```diff
@@ -197,34 +189,29 @@ IEnumerable<NuGetReference> GetTransitiveDependencies()
         public IEnumerable<Reference> References =>
             ProjectRoot.GetAllReferences().Select(r => r.AsReference()).ToList();
 
-        public TargetFrameworkMoniker TFM
+        public IEnumerable<TargetFrameworkMoniker> TargetFrameworks
         {
             get
             {
                 if (IsSdk)
                 {
                     // Currently only supporting non-multi-targeting scenarios
-                    return new TargetFrameworkMoniker(GetSingleTFMProperty("TargetFramework"));
+                    return GetTfms("TargetFramework")
+#pragma warning disable CA1507 // Use nameof to express symbol names
+                        .Concat(GetTfms("TargetFrameworks"))
+#pragma warning restore CA1507 // Use nameof to express symbol names
+                        .Select(t => new TargetFrameworkMoniker(t));
                 }
                 else
                 {
                     // Non-SDK projects should have exactly one TargetFrameworkVersion property
-                    return Context.TfmFactory.GetTFMForNetFxVersion(GetSingleTFMProperty("TargetFrameworkVersion"));
+                    return GetTfms("TargetFrameworkVersion").Select(Context.TfmFactory.GetTFMForNetFxVersion);
                 }
 
-                string GetSingleTFMProperty(string propertyName)
-                {
-                    var targetFrameworkProperties = ProjectRoot.Properties.Where(e => e.Name.Equals(propertyName, StringComparison.Ordinal));
-                    var propCount = targetFrameworkProperties.Count();
-
-                    if (propCount != 1)
-                    {
-                        _logger.LogError("Expected project to have exactly one {PropertyName} property. Found {TargetFrameworkCount} TargetFramework properties for project {Project}.", propertyName, propCount, FilePath);
-                        throw new UpgradeException($"Expected project to have exactly one {propertyName} property. Found {propCount} TargetFramework properties for project {FilePath}.");
-                    }
-
-                    return targetFrameworkProperties.Single().Value;
-                }
+                IEnumerable<string> GetTfms(string propertyName)
```

Comment:
Some users have tried running the tool on project files without target framework properties before. We still want to fail nicely in that scenario. I haven't looked yet to see what happens if this returns an empty enumerable, but it may be worth logging an error here if GetTfms is going to return empty.

Reply:
Side question, is everything mention [here](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build) supported by the assistant?

Reply:
This is a scenario that some of the 1P customers have hit, project files with no target framework properties.

Reply:
@paulomorgado Are you referring to the directory.build.props? I believe it should be as we're using MSBuild to evaluate stuff, but it'd be worth checking. I've created https://github.com/dotnet/upgrade-assistant/issues/387 to track that

Reply:
I've updated the TFM ready check to handle that